src/istio/control/client_context_base.cc

```diff
@@ -48,9 +59,25 @@ CheckOptions GetJustCheckOptions(const TransportConfig& config) {
 
 CheckOptions GetCheckOptions(const TransportConfig& config) {
   auto options = GetJustCheckOptions(config);
-  if (config.has_network_fail_policy() &&
-      config.network_fail_policy().policy() == NetworkFailPolicy::FAIL_CLOSE) {
-    options.network_fail_open = false;
+  if (config.has_network_fail_policy()) {
+    if (config.network_fail_policy().policy() ==
+        NetworkFailPolicy::FAIL_CLOSE) {
+      options.network_fail_open = false;
+    }
+
+    if (0 <= config.network_fail_policy().max_retry()) {
+      options.retries = config.network_fail_policy().max_retry();
```

Comment:
As long as the protobuf defines it as a uint32, then the value returned will always be >= 0.  So currently this if statement is useless.   It could be seen as an extreme form of defensive programming in case the return type ever changed to a signed int... or I could just remove it and unconditionally clobber options.retries.   Thoughts?